CMakeLists.txt

```diff
@@ -81,166 +52,314 @@ if (DEFINED LDFLAGS)
-  project_libdeflate
-  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/libdeflate
-  BUILD_IN_SOURCE 1
-  PREFIX ${CMAKE_CURRENT_SOURCE_DIR}/libdeflate
-  CONFIGURE_COMMAND ""
-  BUILD_COMMAND $(MAKE) -C ${CMAKE_CURRENT_SOURCE_DIR}/libdeflate
-    "CC=${CMAKE_C_COMPILER}"
-    "CFLAGS=-fPIC -O3 ${CMAKE_C_FLAGS}" libdeflate.a
-  INSTALL_COMMAND ""
+# Get the current working branch
+execute_process(
+  COMMAND git rev-parse --abbrev-ref HEAD
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  OUTPUT_VARIABLE GIT_BRANCH
+  OUTPUT_STRIP_TRAILING_WHITESPACE
 )
 
-ExternalProject_Get_Property(project_libdeflate install_dir)
-add_library(libdeflate STATIC IMPORTED)
-
-#include_directories(SYSTEM ${install_dir})
-
-set(libdeflate_location ${install_dir}/libdeflate.a)
-message(STATUS "libdeflate target location is ${libdeflate_location}")
-set_property(TARGET libdeflate PROPERTY IMPORTED_LOCATION ${libdeflate_location})
-add_dependencies(libdeflate project_libdeflate)
-
-## htslib
-ExternalProject_Add(
-  project_htslib
-  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/htslib
-  BUILD_IN_SOURCE 1
-  PREFIX ${CMAKE_CURRENT_SOURCE_DIR}/htslib
-  CONFIGURE_COMMAND autoheader COMMAND autoconf COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/htslib/configure --disable-libcurl --disable-gcs --with-libdeflate
-    "CFLAGS=${MYCFLAGS} -g -Wall -O3 ${CMAKE_C_FLAGS} -I${CMAKE_CURRENT_SOURCE_DIR}/libdeflate"
-    "LDFLAGS=${MYLDFLAGS} -L${CMAKE_CURRENT_SOURCE_DIR}/libdeflate"
-    "CC=${CMAKE_C_COMPILER}"
-  BUILD_COMMAND $(MAKE) -C ${CMAKE_CURRENT_SOURCE_DIR}/htslib libhts.a
-  INSTALL_COMMAND ""
+# Get the latest abbreviated SHA commit hash of the working branch
+execute_process(
+  COMMAND git log -1 --format=%h
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  OUTPUT_VARIABLE GIT_COMMIT_SHORT_HASH
+  OUTPUT_STRIP_TRAILING_WHITESPACE
 )
 
-ExternalProject_Get_Property(project_htslib install_dir)
-add_library(htslib STATIC IMPORTED)
+# Get the latest SHA commit hash of the working branch
+execute_process(
+  COMMAND git rev-parse HEAD
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  OUTPUT_VARIABLE GIT_COMMIT_LONG_HASH
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
 
-# both are needed
-include_directories(SYSTEM ${install_dir})
-include_directories(SYSTEM ${install_dir}/htslib)
+# Get the latest SHA commit hash of the working branch
+execute_process(
+  COMMAND git diff COMMAND wc -l
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  OUTPUT_VARIABLE GIT_NUM_DIRTY_LINES
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
 
-set(htslib_location ${install_dir}/libhts.a)
-message(STATUS "htslib target location is ${htslib_location}")
-set_property(TARGET htslib PROPERTY IMPORTED_LOCATION ${htslib_location})
-add_dependencies(htslib project_htslib)
-add_dependencies(project_htslib libdeflate)
+# configure a header file to pass some of the CMake settings to the source code
+configure_file (
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/graphtyper/constants.hpp.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/include/graphtyper/constants.hpp"
+)
```

Comment:
It would be good to change this section so it works when people build from a tarbal and not from git.